### PR TITLE
feat: add non bnmo purchases to CollectorResume buyerActivity

### PR DESCRIPTION
**Pull request by egdbear**
This PR addresses the calculation of non-BNMO purchases and combines them with BNMO purchases from Exchange.

Please find related Gravity PR [here](https://github.com/artsy/gravity/pull/17444).

[DIA-503]

[DIA-503]: https://artsyproduct.atlassian.net/browse/DIA-503?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/lib/stitching/exchange/v2/stitching.ts [hunk 1]
```diff
@@ -357,21 +357,44 @@ export const exchangeStitchingEnvironment = ({
           fragment: gql`
             ... on CollectorResume {
              userId
+             purchases {
```
Comment (egdbear) [inline src/lib/stitching/exchange/v2/stitching.ts:360]: mb better naming here like `bnmoPurchases` or similar

File: src/schema/v2/conversation/collectorResume.ts [hunk 2]
```diff
@@ -25,5 +26,26 @@ export const CollectorResume = new GraphQLObjectType<any, ResolverContext>({
...
+const CollectorPurchasesType = new GraphQLObjectType<any, ResolverContext>({
+  name: "purchases",
+  fields: {
+    totalAuctionCount: {
+      type: GraphQLInt,
+      description: "Total number of auction winning bids",
+      resolve: ({ auction }) => auction,
+    },
+    totalPrivateSaleCount: {
+      type: GraphQLInt,
```
Comment (dzucconi) [inline src/schema/v2/conversation/collectorResume.ts:46]: Both should be non-nullable

Comment (egdbear) [reply src/schema/v2/conversation/collectorResume.ts]: good one

File: src/schema/v2/conversation/collectorResume.ts [hunk 3]
```diff
@@ -25,5 +26,26 @@ export const CollectorResume = new GraphQLObjectType<any, ResolverContext>({
...
+  fields: {
+    totalAuctionCount: {
+      type: GraphQLNonNull(GraphQLInt),
+      description: "Total number of auction winning bids",
+      resolve: ({ auction }) => auction || 0,
+    },
+    totalPrivateSaleCount: {
+      type: GraphQLNonNull(GraphQLInt),
+      description: "Total number of private sales",
+      resolve: (data) => data["private sale"] || 0,
```
Comment (iskounen) [inline src/schema/v2/conversation/collectorResume.ts:48]: **nitpick:** Can you follow the pattern set above and do `resolve({ private_sale }) => private_sale || 0`?

File: src/lib/stitching/exchange/v2/stitching.ts [hunk 4]
```diff
@@ -357,21 +357,44 @@ export const exchangeStitchingEnvironment = ({
...
+                fieldName: "commerceBuyerActivity",
+                args: exchangeArgs,
+                context,
+                info,
+              })
+
+              return {
+                totalPurchases: bnmoPurchases.totalPurchases + nonBnmoPurchases,
+              }
+            } catch (error) {
```
Comment (iskounen) [inline src/lib/stitching/exchange/v2/stitching.ts:388]: **question:**

1. What kinds of errors are you anticipating? 
2. Why do you want to only return `nonBnmoPurchases` if there is an error?

File: src/lib/stitching/exchange/v2/stitching.ts [hunk 5]
```diff
@@ -357,21 +357,44 @@ export const exchangeStitchingEnvironment = ({
...
+                schema: exchangeSchema,
+                operation: "query",
+                fieldName: "commerceBuyerActivity",
+                args: exchangeArgs,
+                context,
+                info,
+              })
+
+              return {
+                totalPurchases: bnmoPurchases.totalPurchases + nonBnmoPurchases,
```
Comment (iskounen) [inline src/lib/stitching/exchange/v2/stitching.ts:386]: **praise:** I like how easy it is to understand the types of purchases are included in `totalPurchases` when reading this method.

File: src/schema/v2/conversation/collectorResume.ts (hunk 3 above)
Comment (egdbear) [reply src/schema/v2/conversation/collectorResume.ts:48]: Unfortunately not - the field is coming from the [Purchase type "private sale"](https://github.com/artsy/gravity/blob/main/app/models/domain/purchase.rb)

File: src/lib/stitching/exchange/v2/stitching.ts (hunk 4 above)
Comment (egdbear) [reply src/lib/stitching/exchange/v2/stitching.ts:388]: 1. The unauthorised errors (when user without partner authorization access this data for example)
2. I think that was the leftover, and I removed that!

**Review by iskounen — APPROVED** [review]

File: src/schema/v2/conversation/collectorResume.ts [hunk 6]
```diff
@@ -25,5 +26,26 @@ export const CollectorResume = new GraphQLObjectType<any, ResolverContext>({
...
+      resolve: ({ purchases }) => purchases,
+    },
   }),
 })
+
+const CollectorPurchasesType = new GraphQLObjectType<any, ResolverContext>({
+  name: "purchases",
+  fields: {
+    totalAuctionCount: {
+      type: GraphQLNonNull(GraphQLInt),
```
Comment (dzucconi) [inline src/schema/v2/conversation/collectorResume.ts:41]:
```suggestion
      type: new GraphQLNonNull(GraphQLInt),
```

File: src/schema/v2/conversation/collectorResume.ts [hunk 7]
```diff
@@ -25,5 +26,26 @@ export const CollectorResume = new GraphQLObjectType<any, ResolverContext>({
...
+      description: "Total number of auction winning bids",
+      resolve: ({ auction }) => auction || 0,
+    },
+    totalPrivateSaleCount: {
+      type: GraphQLNonNull(GraphQLInt),
```
Comment (dzucconi) [inline src/schema/v2/conversation/collectorResume.ts:46]:
```suggestion
      type: new GraphQLNonNull(GraphQLInt),
```

**Review by dzucconi — DISMISSED** [review]

File: src/lib/stitching/exchange/v2/stitching.ts [hunk 8]
```diff
@@ -357,21 +357,42 @@ export const exchangeStitchingEnvironment = ({
...
+              }
+            } catch (error) {
+              console.error(
+                "[schema/v2/conversation/collectorResume/totalPurchases] Error:",
+                error
+              )
+
+              return { totalPurchases: null }
```
Comment (dzucconi) [inline src/lib/stitching/exchange/v2/stitching.ts:394]: Shouldn't this return 0?

**Pull request merged**